src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -456,7 +470,310 @@ void QMCCostFunctionBatched::engine_checkConfigurations(cqmc::engine::LMYEngine<
                                                         DescentEngine& descentEngineObj,
                                                         const std::string& MinMethod)
 {
-  APP_ABORT("LMYEngine not implemented with batch optimization");
+
```

Comment:
I wonder if there's a way to architect the code to avoid duplicating checkConfigurations into engine_checkConfigurations? The changes between them are not very large. 
The big issue is the parameters that get passed - checkConfigurations doesn't have any, and engine_checkConfigurations needs EngineObj, descentEngineObj, and MinMethod.  
Potential options
1. Combine the three arguments into a single struct (engineData?) that gets added to checkConfigurations.  the contents of this struct could be defined differently based on HAVE_LMY_ENGINE.
The added code inside of checkConfigurations is fairly well separated and could be protected by '#ifdef'
2. Fancier version - create a base class with callbacks that get called at different points in the function.   Pass that class as an argument to checkConfigurations. This is probably overkill for now.

Reply:
Regarding parameters consumed by the optimizes, they are all defined in the top level class and keeps growing. They should be organized in structs.